tools/linter.py

```diff
@@ -50,9 +50,9 @@ def run_check_c_api(self) -> tuple[int, str]:
         # Running borrowed ref checker
         print("Running C API borrow-reference linter...")
         borrowed_ref_script = os.path.join(self.repository_root, "tools", "ci",
-                                           "check_c_api_usage.sh")
+                                           "check_c_api_usage.py")
         borrowed_res = subprocess.run(
-            ["bash", borrowed_ref_script],
+            ["python3", borrowed_ref_script],
```

Comment:
Would it be slightly safer to use `sys.executable` here? I'm reminded of a recent discussion (https://github.com/scipy/scipy/issues/23334) about preferring that in general with `spin`.

Reply:
ok added this `sys.executable` in the script